Testnet/StandardToken/StandardToken/StandardToken.cs

```diff
@@ -163,6 +163,197 @@ public struct ApprovalLog
+    /// </summary>
+    /// <remarks>Emits a TransferLog event with the 'from' address set to the zero address.</remarks>
+    /// <param name="account">The account the newly minted tokens should be credited to.</param>
+    /// <param name="amount">The amount of tokens to mint.</param>
+    void Mint(Address account, UInt256 amount);
+}
+
+/// <summary>
+/// A subset of the ERC20Burnable interface used by OpenZeppelin contracts.
+/// </summary>
+public interface IBurnable
+{
+    /// <summary>
+    /// A user can burn tokens if the have the requisite balance available.
+    /// Burnt tokens are permanently removed from the total supply and cannot be retrieved.
+    /// </summary>
+    /// <remarks>Emits a TransferLog event with the 'to' address set to the zero address.</remarks>
+    /// <param name="amount">The quantity of tokens to be burnt.</param>
+    bool Burn(UInt256 amount);
+}
+
+/// <summary>
+/// An extension of the IBurnable functionality that allows additional data (e.g. for tagging) to be recorded.
+/// </summary>
+public interface IBurnableWithMetadata
+{
+    /// <summary>
+    /// A user can burn tokens if the have the requisite balance available.
+    /// Burnt tokens are permanently removed from the total supply and cannot be retrieved.
+    /// </summary>
+    /// <remarks>Emits a TransferLog event with the 'to' address set to the zero address.
+    /// Additionally emits a BurnMetadata event containing the metadata string.</remarks>
+    /// <param name="amount">The quantity of tokens to be burnt.</param>
+    /// <param name="metadata">Additional data to be recorded with the burn.
+    /// The structure and interpretation of this data is unspecified here.</param>
+    bool BurnWithMetadata(UInt256 amount, string metadata);
+}
+
+[Deploy]
+public class StandardTokenEnhanced : StandardToken, IOwnable, IMintable, IBurnable, IBurnableWithMetadata
+{
+    public StandardTokenEnhanced(ISmartContractState smartContractState, UInt256 totalSupply, string name, string symbol, byte decimals) : base(smartContractState, totalSupply, name, symbol, decimals)
+    {
+        this.Owner = Message.Sender;
+    }
+
+    /// <inheritdoc />
+    public Address Owner
+    {
+        get => State.GetAddress(nameof(this.Owner));
+        private set => State.SetAddress(nameof(this.Owner), value);
+    }
+
+    /// <inheritdoc />
+    public void OnlyOwner()
+    {
+        Assert(IsOwner(), "Only the owner can call this method");
+    }
+
+    /// <inheritdoc />
+    public bool IsOwner()
+    {
+        return Message.Sender == Owner;
+    }
+
+    /// <inheritdoc />
+    public void RenounceOwnership()
+    {
+        OnlyOwner();
+
+        Address previousOwner = this.Owner;
+
+        this.Owner = Address.Zero;
+
+        Log(new OwnershipTransferred { PreviousOwner = previousOwner, NewOwner = Address.Zero });
+    }
+
+    /// <inheritdoc />
+    public void TransferOwnership(Address newOwner)
```

Comment:
A certiK audit will tell you to assign a PendingOwner, then another method to have that PendingOwner come in and ClaimOwnership to avoid pushing ownership to a potentially invalid address.